src/schemas/common.ts

```diff
@@ -0,0 +1,134 @@
+import { z } from "zod";
+import * as constants from "../constants";
+import {
+  AppendExpectedRevision,
+  BinaryEventType,
+  Credentials,
+  EventType,
+  ExpectedRevision,
+  Filter,
+  FilterBase,
+  FilterOn,
+  JSONEventType,
+  JSONType,
+  MetadataType,
+  Position,
+  ReadPosition,
+  ReadRevision,
+  RegexFilter,
+} from "../types";
+
+export const credentials = z.object({
+  username: z.string(),
+  password: z.string(),
+}) satisfies z.ZodType<Credentials>;
+
+export const baseOptions = z.object({
+  credentials: credentials.optional(),
+  requiresLeader: z.boolean().optional(),
+  deadline: z.number().min(0).optional(),
+}) satisfies z.ZodType;
+
+export const jsonType = z.union([
+  z.record(z.union([z.string(), z.number()]), z.unknown()),
+  z.array(z.unknown()),
+  z.string(),
+]) satisfies z.ZodType<JSONType>;
+
+export const metadataType = z.union([
+  jsonType,
```

Comment:
Same here 🤔 I would expect this to be inlined with other SDKs where the `metadata` is a `byte[]` of anything